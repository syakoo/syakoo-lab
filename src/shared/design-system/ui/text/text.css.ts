import { styleVariants } from "@vanilla-extract/css";
import { recipe } from "@vanilla-extract/recipes";

import { theme } from "@/shared/design-system/theme.css";

const colors = {
  ...theme.color.text,
  currentColor: "currentColor",
} as const;

export const textStyle = recipe({
  base: {
    lineHeight: 1.5,
  },

  variants: {
    // TODO: 詳細はパス
    color: styleVariants(colors, (color) => ({
      color,
    })),
    size: styleVariants(theme.fontSize, (size) => ({
      fontSize: size,
    })),
    weight: {
      bold: {
        fontWeight: "bold",
      },
      normal: {
        fontWeight: "normal",
      },
    },
  },
});
