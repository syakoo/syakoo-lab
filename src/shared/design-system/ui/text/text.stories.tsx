import type { Meta } from "@storybook/nextjs";

import { H1, H2, H3, H4, P, Span, Text } from ".";

const meta = {
  component: Text,
  tags: ["autodocs"],
  parameters: {},
} satisfies Meta<typeof Text>;

export default meta;

export const Sample = () => {
  return (
    <>
      <H1>H1</H1>
      <H2>H2</H2>
      <H3>H3</H3>
      <H4>H4</H4>
      <P>P</P>
      <Span>Span</Span>
    </>
  );
};

export const Colors = () => {
  return (
    <>
      <Text color="currentColor">currentColor (default)</Text>
      <Text color="primary">primary</Text>
      <Text color="secondary">secondary</Text>
      <Text color="tertiary">tertiary</Text>
    </>
  );
};

export const Sizes = () => {
  return (
    <>
      <Text size="50">50</Text>
      <Text size="100">100 (default)</Text>
      <Text size="200">200</Text>
      <Text size="300">300</Text>
      <Text size="400">400</Text>
      <Text size="500">500</Text>
      <Text size="600">600</Text>
      <Text size="700">700</Text>
      <Text size="800">800</Text>
    </>
  );
};

export const Weights = () => {
  return (
    <>
      <Text size="100" weight="normal">
        normal (default)
      </Text>
      <Text weight="bold">bold</Text>
    </>
  );
};
