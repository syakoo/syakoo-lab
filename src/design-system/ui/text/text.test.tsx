import { testStories } from "@/test/test-stories";

import * as stories from "./text.stories";

describe("Text", () => {
  testStories(stories);
});
