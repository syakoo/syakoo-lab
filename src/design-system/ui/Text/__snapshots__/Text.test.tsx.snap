// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders Colors 1`] = `
<body>
  <div>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      currentColor (default)
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_primary__jzpfqdf Text_textStyle_variants_color_primary__jzpfqd0 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      primary
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      secondary
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_tertiary__jzpfqdh Text_textStyle_variants_color_tertiary__jzpfqd2 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      tertiary
    </p>
  </div>
</body>
`;

exports[`Text renders Sample 1`] = `
<body>
  <div>
    <h1
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_600__jzpfqdq Text_textStyle_variants_size_600__jzpfqdb Text_textStyle_weight_bold__jzpfqdt"
    >
      H1
    </h1>
    <h2
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_500__jzpfqdp Text_textStyle_variants_size_500__jzpfqda Text_textStyle_weight_bold__jzpfqdt"
    >
      H2
    </h2>
    <h3
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_400__jzpfqdo Text_textStyle_variants_size_400__jzpfqd9 Text_textStyle_weight_bold__jzpfqdt"
    >
      H3
    </h3>
    <h4
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_300__jzpfqdn Text_textStyle_variants_size_300__jzpfqd8 Text_textStyle_weight_bold__jzpfqdt"
    >
      H4
    </h4>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      P
    </p>
    <span
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      Span
    </span>
  </div>
</body>
`;

exports[`Text renders Sizes 1`] = `
<body>
  <div>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
    >
      50
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      100 (default)
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_200__jzpfqdm Text_textStyle_variants_size_200__jzpfqd7 Text_textStyle_weight_normal__jzpfqdu"
    >
      200
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_300__jzpfqdn Text_textStyle_variants_size_300__jzpfqd8 Text_textStyle_weight_normal__jzpfqdu"
    >
      300
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_400__jzpfqdo Text_textStyle_variants_size_400__jzpfqd9 Text_textStyle_weight_normal__jzpfqdu"
    >
      400
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_500__jzpfqdp Text_textStyle_variants_size_500__jzpfqda Text_textStyle_weight_normal__jzpfqdu"
    >
      500
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_600__jzpfqdq Text_textStyle_variants_size_600__jzpfqdb Text_textStyle_weight_normal__jzpfqdu"
    >
      600
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_700__jzpfqdr Text_textStyle_variants_size_700__jzpfqdc Text_textStyle_weight_normal__jzpfqdu"
    >
      700
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_800__jzpfqds Text_textStyle_variants_size_800__jzpfqdd Text_textStyle_weight_normal__jzpfqdu"
    >
      800
    </p>
  </div>
</body>
`;

exports[`Text renders Weights 1`] = `
<body>
  <div>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_normal__jzpfqdu"
    >
      normal (default)
    </p>
    <p
      class="Text__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_bold__jzpfqdt"
    >
      bold
    </p>
  </div>
</body>
`;
